app/src/main/java/com/infomaniak/mail/utils/Extensions.kt

```diff
@@ -369,20 +370,34 @@ fun List<Message>.getFoldersIds(exception: String? = null) = mapNotNull { if (it
 fun List<Message>.getUids(): List<String> = map { it.uid }
 //endregion
 
-fun Fragment.deleteWithConfirmationPopup(folderRole: FolderRole?, count: Int, callback: () -> Unit) {
-    when (folderRole) {
-        FolderRole.DRAFT, FolderRole.SPAM, FolderRole.TRASH -> {
-            createDescriptionDialog(
-                title = resources.getQuantityString(R.plurals.threadListDeletionConfirmationAlertTitle, count, count),
-                description = resources.getQuantityString(R.plurals.threadListDeletionConfirmationAlertDescription, count),
-                onPositiveButtonClicked = callback,
-                displayLoader = false
-            ).show()
-        }
-        else -> callback()
+fun Fragment.deleteWithConfirmationPopup(
+    folderRole: FolderRole?,
+    count: Int,
+    displayLoader: Boolean = true,
+    onDismiss: (() -> Unit)? = null,
+    callback: () -> Unit,
+): AlertDialog? = when (folderRole) {
+    FolderRole.DRAFT, FolderRole.SPAM, FolderRole.TRASH -> {
+        createDescriptionDialog(
+            title = resources.getQuantityString(R.plurals.threadListDeletionConfirmationAlertTitle, count, count),
+            description = resources.getQuantityString(R.plurals.threadListDeletionConfirmationAlertDescription, count),
+            displayLoader = displayLoader,
+            onPositiveButtonClicked = callback,
+            onDismissed = onDismiss,
+        ).also { it.show() }
+    }
+    else -> {
+        callback()
+        null
     }
 }
 
+inline var Fragment.deleteThreadDialog
+    get() = (activity as? MainActivity)?.deleteThreadDialog
+    set(value) {
+        (activity as? MainActivity)?.deleteThreadDialog = value
+    }
```

Comment:
Can we share the dialog instance using the main view model instead of the activity?

Reply:
I don't think we want UI stuff like this in a viewModel

Reply:
@sirambd do you have any opinion on the matter?

Reply:
a priori the logic will change in another PR so we can mergeà priori la logique va changer dans une autre PR donc on peut merger